aten/src/ATen/test/atest.cpp

```diff
@@ -185,7 +185,9 @@ TEST_F(atest, ne_operators) {
 
 TEST_F(atest, add_operators) {
   auto exp_tensor = tensor({-10, 1, 0, -1, 10});
-  run_binary_ops_test(add_out, x_tensor, y_tensor, exp_tensor, INTBOOL, 2);
+  run_binary_ops_test<
+      at::Tensor& (*)(at::Tensor&, const at::Tensor&, const at::Tensor&, const at::Scalar&)>(
+      add_out, x_tensor, y_tensor, exp_tensor, INTBOOL, 2);
```

Comment:
`at::add_out` is ambiguous between the existing `add.Tensor_out` and the newly generated `add.Scalar_out`. Hopefully not too BC-breaking? (I guess I should import to phabricator to be safe :(

Reply:
This is not a matter of being BC breaking; this is a matter of `add_out` working at all haha.

So what exactly is going on here. Post this PR, we now have:

```
add.out(Tensor self, Tensor other, *, Scalar alpha=1, Tensor(a!) out) -> Tensor(a!)
add.Scalar_out(Tensor self, Scalar other, Scalar alpha=1, *, Tensor(a!) out) -> Tensor(a!)
``

giving

```
add_out(Tensor& out, const Tensor& self, const Tensor& other, const Scalar& alpha = 1)
add_out(Tensor& out, const Tensor& self, const Scalar& other, const Scalar& alpha)
``

I admit, I don't see why this is ambiguous. What's the error message you go there?

Reply:
The error looks like:
```
/raid/hirsheybar/pytorch-functionalization2/aten/src/ATen/test/atest.cpp: In member function ‘virtual void atest_add_operators_Test::TestBody()’:
/raid/hirsheybar/pytorch-functionalization2/aten/src/ATen/test/atest.cpp:189:58: error: no matching function for call to ‘run_binary_ops_test(<unresolved overloaded function type>, at::Tensor&, at::Tensor&, at::Tensor&, const int&, int)’
  189 |       add_out, x_tensor, y_tensor, exp_tensor, INTBOOL, 2);
      |                                                          ^
/raid/hirsheybar/pytorch-functionalization2/aten/src/ATen/test/atest.cpp:76:6: note: candidate: ‘template<class T, class ... Args> void run_binary_ops_test(T, const at::Tensor&, const at::Tensor&, const at::Tensor&, int, Args ...)’
   76 | void run_binary_ops_test(
      |      ^~~~~~~~~~~~~~~~~~~
/raid/hirsheybar/pytorch-functionalization2/aten/src/ATen/test/atest.cpp:76:6: note:   template argument deduction/substitution failed:
/raid/hirsheybar/pytorch-functionalization2/aten/src/ATen/test/atest.cpp:189:58: note:   couldn’t deduce template parameter ‘T’
  189 |       add_out, x_tensor, y_tensor, exp_tensor, INTBOOL, 2);
```

It looks like we're passing `add_out` to a template that expects to know what the type of the first argument is, but it can't deduce it now that there are multiple overloads for `add_out`?